contracts/custom/DecentDAOVoting.sol

```diff
@@ -0,0 +1,111 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity =0.8.19;
+
+import { IVotes } from "@openzeppelin/contracts/governance/utils/IVotes.sol";
+import { LinearERC20Voting } from "../azorius/LinearERC20Voting.sol";
+
+/**
+ * A "token vester" contract, intended to allow for holding an ERC-20 token, to be released under a
+ * predetermined vesting schedule.
+ */
+interface TokenVester {
+
+    /**
+     * Returns the current token balance in the vesting contract, for the given token holder, at a specific
+     * block.
+     *
+     * Timestamping token balances at specific blocks is required on the vesting contract, in order to be
+     * able to use a holder's historical balance when voting on proposals in the voting strategy.  Otherwise
+     * it would be possible to acquire additional votes during the voting of a proposal.
+     *
+     * @param _token the ERC-20 token address for the asset you would like the balance of, in our usage this
+     *          is always the Decent governance token address
+     * @param _holder the token holder's address
+     * @param _block the specific block number to query the balance at
+     */
+    function getBalance(address _token, address _holder, uint _block) external view returns (uint);
+}
+
+interface IVotesDecent is IVotes {
+    /**
+     * Returns the token balance that `account` had at a specific moment in the past.
+     * This is distinct from `getPastVotes` in that it is the account's *balance*, not the number of
+     * delegated token votes, which can be different.
+     */
+    function getPastBalance(address account, uint256 timepoint) external view returns (uint256);
```

Comment:
@adamgall this encapsulates whether to use delegation or balance into the strategy, but requires an additional function on the governance token.